include/nbl/builtin/hlsl/bxdf/common.hlsl

```diff
@@ -48,358 +174,738 @@ struct Basic
+    static this_t create(NBL_CONST_REF_ARG(vector3_type) tangentSpaceV, NBL_CONST_REF_ARG(vector3_type) tangentSpaceH)
+    {
+        this_t retval;
+
+        retval.iso_cache.VdotH = nbl::hlsl::dot<vector3_type>(tangentSpaceV,tangentSpaceH);
+        retval.iso_cache.LdotH = retval.iso_cache.getVdotH();
+        retval.iso_cache.NdotH = tangentSpaceH.z;
+        retval.iso_cache.NdotH2 = retval.iso_cache.getNdotH() * retval.iso_cache.getNdotH();
+        retval.TdotH = tangentSpaceH.x;
+        retval.BdotH = tangentSpaceH.y;
+
+        return retval;
     }
-    
-    return retval;
-  }
-  // always valid because its specialized for the reflective case
-  static AnisotropicMicrofacetCache createForReflection(const float3 tangentSpaceV, const float3 tangentSpaceL, const float VdotL)
-  {
-    AnisotropicMicrofacetCache retval;
-    
-    float LplusV_rcpLen;
-    retval = createForReflection(tangentSpaceV.z,tangentSpaceL.z,VdotL,LplusV_rcpLen);
-    retval.TdotH = (tangentSpaceV.x+tangentSpaceL.x)*LplusV_rcpLen;
-    retval.BdotH = (tangentSpaceV.y+tangentSpaceL.y)*LplusV_rcpLen;
-    
-    return retval;
-  }
-  template<class ObserverRayDirInfo, class IncomingRayDirInfo>
-  static AnisotropicMicrofacetCache createForReflection(
-    const surface_interactions::Anisotropic<ObserverRayDirInfo> interaction, 
-    const LightSample<IncomingRayDirInfo> _sample)
-  {
-    return createForReflection(interaction.getTangentSpaceV(),_sample.getTangentSpaceL(),_sample.VdotL);
-  }
-  // transmissive cases need to be checked if the path is valid before usage
-  static bool compute(
-    out AnisotropicMicrofacetCache retval,
-    const bool transmitted, const float3 V, const float3 L,
-    const float3 T, const float3 B, const float3 N,
-    const float NdotL, const float VdotL,
-    const float orientedEta, const float rcpOrientedEta, out float3 H
-  )
-  {
-    float3 H;
-    const bool valid = IsotropicMicrofacetCache::compute(retval,transmitted,V,L,N,NdotL,VdotL,orientedEta,rcpOrientedEta,H);
-    if (valid)
-    {
-      retval.TdotH = dot(T,H);
-      retval.BdotH = dot(B,H);
-    }
-    return valid;
-  }
-  template<class ObserverRayDirInfo, class IncomingRayDirInfo>
-  static bool compute(
-    out AnisotropicMicrofacetCache retval,
-    const surface_interactions::Anisotropic<ObserverRayDirInfo> interaction, 
-    const LightSample<IncomingRayDirInfo> _sample,
-    const float eta
-  )
-  {
-    float3 H;
-    const bool valid = IsotropicMicrofacetCache::compute(retval,interaction,_sample,eta,H);
-    if (valid)
-    {
-      retval.TdotH = dot(interaction.T,H);
-      retval.BdotH = dot(interaction.B,H);
-    }
-    return valid;
-  }
-
-  float TdotH;
-  float BdotH;
+    static this_t create(
+        NBL_CONST_REF_ARG(vector3_type) tangentSpaceV,
+        NBL_CONST_REF_ARG(vector3_type) tangentSpaceH,
+        const bool transmitted,
+        const scalar_type rcpOrientedEta,
+        const scalar_type rcpOrientedEta2
```

Comment:
you already have a struct for this https://github.com/Devsh-Graphics-Programming/Nabla/blob/deda14aa0d7fe0061fc0a876d665b5686e74e0bd/include/nbl/builtin/hlsl/bxdf/fresnel.hlsl#L54C8-L54C23

(from which I told you to remove the `backside` boolean member)